src/server.ts

```diff
@@ -32,6 +44,294 @@ app.get("/health-check", async (_req, res) => {
     }
 });
 
+// ROUTE HANDLERS: /users
+app.get<{}, User[] | string>("/users", async (_req, res) => {
+    try {
+        const result = await queryAndLog(client, "SELECT * FROM users");
+        res.status(200).json(result.rows);
+    } catch (error) {
+        console.error(error);
+        res.status(500).send("An error occurred. Check server logs.");
+    }
+});
+
+app.post<{}, User | string, UserCandidate>("/users", async (req, res) => {
+    try {
+        const result = await queryAndLog(
+            client,
+            "INSERT INTO users (name) VALUES ($1) RETURNING *",
+            [req.body.name]
+        );
+        result.rowCount === 1
+            ? res.status(201).json(result.rows[0])
+            : res.status(400).send("An error occurred. Check server logs.");
+    } catch (error) {
+        console.error(error);
+        res.status(500).send("An error occurred. Check server logs.");
+    }
+});
+
+// ROUTE HANDLERS: /decks
+app.get<{}, Deck[] | string>("/decks", async (_req, res) => {
+    try {
+        const result = await queryAndLog(client, "SELECT * FROM decks");
+        res.status(200).json(result.rows);
+    } catch (error) {
+        console.error(error);
+        res.status(500).send("An error occurred. Check server logs.");
+    }
+});
+
+app.post<{}, Deck | string, DeckCandidate>("/decks", async (req, res) => {
+    try {
+        const result = await queryAndLog(
+            client,
+            "INSERT INTO decks (name) VALUES ($1) RETURNING *",
+            [req.body.name]
+        );
+        result.rowCount === 1
+            ? res.status(201).json(result.rows[0])
+            : res.status(400).send("Deck not created.");
+    } catch (error) {
+        console.error(error);
+        res.status(500).send("An error occurred. Check server logs.");
+    }
+});
+
+app.patch<{ id: string }, Deck[] | string, DeckCandidate>(
+    "/decks/:id",
+    async (req, res) => {
+        try {
+            const result = await queryAndLog(
+                client,
+                "UPDATE decks SET name = $2 WHERE id = $1 RETURNING *",
+                [req.params.id, req.body.name]
+            );
+
+            result.rowCount === 1
+                ? res.status(200).json(result.rows)
+                : res.status(204).send();
```

Comment:
what's the purpose of that? what's the scenario where the second options runs? (just a question)

Reply:
In the event that a select statement returns no rows (nothing there) we send a different response (204 means "no content")